be/src/runtime/task_group/task_group_manager.cpp

```diff
@@ -69,18 +69,16 @@ TaskGroupPtr TaskGroupManager::get_task_group_by_id(uint64_t tg_id) {
 
 bool TaskGroupManager::set_cg_task_sche_for_query_ctx(uint64_t tg_id, QueryContext* query_ctx_ptr) {
     std::lock_guard<std::shared_mutex> write_lock(_task_scheduler_lock);
-    if (_tg_sche_map.find(tg_id) != _tg_sche_map.end()) {
-        query_ctx_ptr->set_task_scheduler(_tg_sche_map.at(tg_id).get());
-    } else {
-        return false;
-    }
```

Comment:
The original one can support failing fast and it can be more efficient than the modifications under some scenarios.

Reply:
> The original one can support failing fast and it can be more efficient than the modifications under some scenarios.
The process for failure scenarios is the same